subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -1694,6 +1735,18 @@ int nrf_cloud_shadow_dev_status_encode(const struct nrf_cloud_device_status *con
 		parent_obj = cJSON_AddObjectToObjectCS(root_obj, NRF_CLOUD_JSON_KEY_REP);
 	}
 
+	cJSON *connection_obj = cJSON_AddObjectToObjectCS(parent_obj, NRF_CLOUD_JSON_KEY_CONN);
```

Comment:
since this function is for adding the device status as indicated by the provided struct, can you wrap the entire thing:
```
#if defined(CONFIG_NRF_CLOUD_REPORT_CONNECTION)
        cJSON *connection_obj = cJSON_AddObjectToObjectCS(parent_obj, NRF_CLOUD_JSON_KEY_CONN);
    
	if (connection_obj == NULL) {
		err = -ENOMEM;
		goto cleanup;
	}

	err = shadow_connection_info_update(connection_obj);
	if (err) {
		goto cleanup;
	}
#endif
```

or maybe it would make sense to add it as on option in the struct?

Reply:
This function is called by `nrf_cloud_shadow_device_status_update()` which is in turn called by a couple of different MQTT-based samples and applications.  I would like to leave it as is.

Reply:
yes, my point is that you are adding new functionality to this function which is different from its original purpose.
it originally encoded device status as requested by the provided struct.
now it is adding a connection object.

Reply:
True.  Maybe it would be better to report it inside device in the shadow:
```
    "reported": {
      "device": {
        "deviceInfo": {
          "modemFirmware": "mfw_nrf9160_1.3.5"
        },
        "serviceInfo": {
        ...
        },
        "connInfo": {
                "protocol": "MQTT",
                "method": "LTE"
         }
      }
   }

Reply:
i like that for this feature.

Reply:
consider squashing in the following commit: 
https://github.com/nrfconnect/sdk-nrf/commit/35564b1676bfb6dfc0f0bdbf0e6940859c924fe4

it was easier than explaining 😅

Reply:
Thanks for doing the work.  I pulled it in, rebased / split / etc., and added use of it to the other samples.